Modules/posixmodule.c

```diff
@@ -5309,52 +5311,68 @@ os_spawnve_impl(PyObject *module, int mode, path_t *path, PyObject *argv,
 
 
 #ifdef HAVE_FORK
+
+/* Helper function to validate arguments.
+   Returns 0 on success.  1 on failure with a TypeError raised.
+   If obj is non-NULL it must be callable.  */
+static int
+check_null_or_callable(PyObject *obj, const char* obj_name)
+{
+    if (obj && !PyCallable_Check(obj)) {
+        PyErr_Format(PyExc_TypeError, "'%s' must be callable, not %R",
+                     obj_name, Py_TYPE(obj));
+        return 1;
+    }
+    return 0;
+}
+
 /*[clinic input]
 os.register_at_fork
 
-    func: object
-        Function or callable
-    /
-    when: str
-        'before', 'child' or 'parent'
+    *
+    before: object=NULL
+        Zero arg callable to be called in the parent before the fork() syscall.
+    after_in_child: object=NULL
+        Zero arg callable to be called in the child before os.fork() returns.
+    after_in_parent: object=NULL
+        Zero arg callable to be called in the parent before os.fork() returns.
 
-Register a callable object to be called when forking.
+Register a callable to be called when forking a new process via os.fork().
 
 'before' callbacks are called in reverse order before forking.
```

Comment:
This partially repeats descriptions of arguments ('before' is called before forking, etc).